src/Connections/GenericHook.ts

```diff
@@ -258,18 +264,41 @@ export class GenericHookConnection extends BaseConnection implements IConnection
         }
     }
 
-    public async onStateUpdate(stateEv: MatrixEvent<unknown>) {
-        const validatedConfig = GenericHookConnection.validateState(stateEv.content as Record<string, unknown>, this.config.allowJsTransformationFunctions || false);
-        if (validatedConfig.transformationFunction) {
-            try {
-                this.transformationFunction = new Script(validatedConfig.transformationFunction);
-            } catch (ex) {
-                await this.messageClient.sendMatrixText(this.roomId, 'Could not compile transformation function:' + ex, "m.text", this.intent.userId);
-            }
+    public onStateUpdate(stateEv: MatrixEvent<unknown>) {
+        return this.updateConfig(stateEv.content as Record<string, unknown>);
```

Comment:
It feels like updateConfig could actually be in here, this function isn't achieving much.

Reply:
This is here to wrap around `updateConfig`, which is also called by `provisionerUpdateConfig`.  i.e. a config update can come from either a direct update to its room state event, or from a call to the provisioning API; this light `onStateUpdate` is just a thin layer to make `updateConfig` compatible with `IConnectionState#onStateUpdate`.